core/src/services/webhdfs/writer.rs

```diff
@@ -41,14 +43,67 @@ impl WebhdfsWriter {
 }
 
 #[async_trait]
-impl oio::OneShotWrite for WebhdfsWriter {
+impl oio::MultipartUploadWrite for WebhdfsWriter {
```

Comment:
Implementing `MultipartUpload` for `WebhdfsWriter` seems not fine to me. It lacks an `upload_id` and cannot be aborted using an upload ID.

It's behavior is more like https://github.com/apache/incubator-opendal/pull/3769. Maybe we can add an `oio::BlockWrite`:

- We can write blocks with block_id we generated
- We can compose blocks with a block_lists

Would you like to implement `oio::BlockWrite`?

Reply:
I am very willing.

Reply:
> Should I new a pr to do this?

Yes, it would be better to have a separate PR for it.